### PR TITLE
[Feature] Implement side navbar

### PR DESCRIPTION
**Pull request by ABCodez**
Navbar should be completely responsive! Any feedback/changed would be appreciated :)

image
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,8 +20,10 @@
         "@types/node": "^20.6.0",
         "class-variance-authority": "^0.7.0",
         "firebase": "^10.5.2",
+        "hamburger-react": "^2.5.0",
         "react": "^18.2.0",
         "react-dom": "^18.2.0",
+        "react-icons": "^5.0.1",
```
Comment (juancwu) [inline package.json:26]: Was there an icon that couldn't be found in `@heroicons/react`? We already have an icon package, just don't want to have duplicates

File: src/components/Navbar.tsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        };
+    }, []);
+
+    const renderNavItems = (isMobile: boolean) =>
+        navItems.map(({ path, label, Icon }) => (
+            <li
+                key={label}
+                className="p-4 hover:bg-slate-100 duration-300 transition-colors rounded-md w-full hover:text-black"
+            >
+                <a
```
Comment (juancwu) [inline src/components/Navbar.tsx:40]: The `<a`> tags need to be replace with the `<Link>` component from `react-router-dom`. The native tag causes the page to do a reload, while the `<Link>` does a repaint with the react app that has already been downloaded.

**Review by juancwu — CHANGES_REQUESTED** [review]

File: src/components/Navbar.tsx (hunk 2 above)
Comment (ABCodez) [reply src/components/Navbar.tsx]: resolved in newest commit

File: package.json (hunk 1 above)
Comment (ABCodez) [reply package.json:26]: The icons from heroicons were either missing or way too different compared to the Figma file. I moved to using react-icons for that reason, but I could extract the icon SVGs from the Figma file and use those if need be!

Comment (juancwu) [reply package.json:26]: @ABCodez Its good then, just wanted to know why we needed another icons package. We can leave it as is.

**Review by juancwu — APPROVED** [review]

File: src/components/Navbar.tsx [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+                        </div>
+                    </nav>
+
+                    <div
+                        className={`fixed right-0 top-0 z-40 h-full w-full max-w-[65%] p-10 py-24 bg-gray-200 backdrop-blur-xl transition-transform duration-300 ease-in-out ${
+                            isMobileMenuOpen
+                                ? "translate-x-0 opacity-100"
+                                : "translate-x-full opacity-0"
+                        } `}
+                    >
```
Comment (AmirAgassi) [inline src/components/Navbar.tsx:91]:
```suggestion
                        <div
                            className={`fixed right-0 top-0 z-40 h-full w-full max-w-[12.5rem] p-10 py-24 bg-gray-200 backdrop-blur-xl transition-all duration-300 ease-in-out ${
                                isMobileMenuOpen ? "translate-x-0 opacity-100" : "translate-x-full opacity-0"
                            }`}
                        >
```

image
Should it use % for the width of the mobile bar? Could look weird near the upper end of mobile sizes.

Also copied over the smooth opening animation to the closing of the menu as well.

Everything else looks awesome 🔥

Comment (ABCodez) [reply src/components/Navbar.tsx]: I noticed this too! Your changes make this look wayyy better

**Review by juancwu — APPROVED** [review]

**Pull request merged**